src/librustc_incremental/calculate_svh/hasher.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::hash::{Hasher, SipHasher};
+
+#[derive(Debug)]
+pub struct IchHasher {
+    // FIXME: this should use SHA1, not SipHash. SipHash is not
+    // built to avoid collisions.
+    state: SipHasher,
+    bytes_hashed: u64,
+}
+
+impl IchHasher {
+
```

Comment:
Most of the other impls in this file (and across the compiler) don't have it. "Ask the style team." (not that there are concrete answers there at this point either).